src/main/java/ArrayOperations.java

```diff
@@ -37,58 +54,85 @@ public void print() {
      * @see <a href="sorting algortihms">http://faculty.cs.niu.edu/~hutchins/csci241/sorting.htm</a>
      */
     public int[] revertSort() {
-        return null;
+
+       int[] numb = sort();
+        numb = reverse(numb);
+
+        return numb;
     }
 
     /**
      * @return the unsorted array in reverted order.
      */
     public int[] reverted() {
-        return null;
+
+        return reverse(this.numbers);
+
     }
 
     /**
      * @param value which should be searched for.
      * @return true if the array contains the value, false otherwise.
      */
     public boolean contains(int value) {
+        for(int i=0; i < numbers.length; i++) {
+            if (value == numbers[i]) {
+                return true;
+            }
+        }
         return false;
     }
 
     /**
      * @return the average value of all elements summed up.
      */
     public double average() {
-        return 0.0;
+        double sum=0;
+        for ( int i=0; i < numbers.length; i++) {
+            sum = sum + numbers[i];
+        }
+        return sum / numbers.length;
     }
 
     /**
      * @return the average value of all elements summed up, but without the highest and the lowest value.
      */
     public double trimmedMean() {
-        return 0.0;
+        int [] numbers= sort();
+        double sum=0;
+        for ( int i=1; i < numbers.length-1; i++) {
+            sum = sum + numbers[i];
+        }
+
+        return sum / numbers.length-1;
```

Comment:
-2 since you cut off two elements (the smallest and the greatest). You also need to make brackets.
`return sum / (numbers.length - 2);` Otherwise good solution.